drools/src/main/java/com/datatorrent/drools/rules/AbstractRulesReader.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2017 DataTorrent, Inc. ALL Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.drools.rules;
+
+import java.io.IOException;
+
+import org.apache.hadoop.classification.InterfaceStability.Evolving;
+
+@Evolving
+public interface AbstractRulesReader
+{
+
+  /**
+   * Load rules files from directory
+   * @param rulesDir
+   * @return kieFileSystem
+   * @throws IOException
+   */
+  void loadRulesFromDirectory(String rulesDir) throws IOException;
```

Comment:
I was wondering, what if someone doesn't want to use HadoopFileSystem and something else. If we pass Path, that subclass will have to retrieve path string and then do needful. May be I can change private functions param type to Path. Does that sound okay?